src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -226,5 +226,90 @@ public static object InvokeMethod( MethodInfo method, object fixture, params obj
+        /// rather than throwing <see cref="AmbiguousMatchException"/>
+        /// for properties that shadow properties of a different property type.
+        /// </para>
+        /// <para>
+        /// If you request both public and nonpublic properties, every public property is preferred
+        /// over every nonpublic property. It would violate the principle of least surprise for a
+        /// derived class�s implementation detail to be chosen over the public API for a type.
+        /// </para>
+        /// </summary>
+#elif NETSTANDARD1_6
+        /// <summary>
+        /// <para>
+        /// Selects the ultimate shadowing property just like <see langword="dynamic"/> would,
+        /// rather than throwing <see cref="AmbiguousMatchException"/>
+        /// for properties that shadow properties of a different property type
+        /// which is what <see cref="TypeInfo.GetProperty(string, BindingFlags)"/> does.
+        /// </para>
+        /// <para>
+        /// If you request both public and nonpublic properties, every public property is preferred
+        /// over every nonpublic property. It would violate the principle of least surprise for a
+        /// derived class�s implementation detail to be chosen over the public API for a type.
+        /// </para>
+        /// </summary>
+        /// <param name="type">See <see cref="TypeInfo.GetProperty(string, BindingFlags)"/>.</param>
+        /// <param name="name">See <see cref="TypeInfo.GetProperty(string, BindingFlags)"/>.</param>
+        /// <param name="bindingFlags">See <see cref="TypeInfo.GetProperty(string, BindingFlags)"/>.</param>
+#else
+        /// <summary>
+        /// <para>
+        /// Selects the ultimate shadowing property just like <see langword="dynamic"/> would,
+        /// rather than throwing <see cref="AmbiguousMatchException"/>
+        /// for properties that shadow properties of a different property type
+        /// which is what <see cref="Type.GetProperty(string, BindingFlags)"/> does.
+        /// </para>
+        /// <para>
+        /// If you request both public and nonpublic properties, every public property is preferred
+        /// over every nonpublic property. It would violate the principle of least surprise for a
+        /// derived class�s implementation detail to be chosen over the public API for a type.
+        /// </para>
+        /// </summary>
+        /// <param name="type">See <see cref="Type.GetProperty(string, BindingFlags)"/>.</param>
+        /// <param name="name">See <see cref="Type.GetProperty(string, BindingFlags)"/>.</param>
+        /// <param name="bindingFlags">See <see cref="Type.GetProperty(string, BindingFlags)"/>.</param>
+#endif
+
+        public static PropertyInfo GetUltimateShadowingProperty(Type type, string name, BindingFlags bindingFlags)
+        {
+            if (type == null) throw new ArgumentNullException(nameof(type));
+            if (name == null) throw new ArgumentNullException(nameof(name));
```

Comment:
How about `Guard.IfNull()`? For consistent handling. 🙂

Reply:
Good idea! Forgot about them.